text/1099-renderComponent.md

```diff
@@ -0,0 +1,179 @@
+---
+stage: accepted
+start-date: 2025-05-01T00:00:00.000Z
+release-date: # In format YYYY-MM-DDT00:00:00.000Z
+release-versions:
+teams: # delete teams that aren't relevant
+  - framework
+prs:
+  accepted: https://github.com/emberjs/rfcs/pull/1099 
+project-link:
+suite: 
+---
+
+<!--- 
+Directions for above: 
+
+stage: Leave as is
+start-date: Fill in with today's date, 2032-12-01T00:00:00.000Z
+release-date: Leave as is
+release-versions: Leave as is
+teams: Include only the [team(s)](README.md#relevant-teams) for which this RFC applies
+prs:
+  accepted: Fill this in with the URL for the Proposal RFC PR
+project-link: Leave as is
+suite: Leave as is
+-->
+
+<!-- Replace "RFC title" with the title of your RFC -->
+
+# `renderComponent` 
+
+## Summary
+
+Add a new API to for rendering components into any DOM element.
+
+## Motivation
+
+Building an app from scratch requires a fair amount of boilerplate today, and `renderComponent` abstracts a minimal API that would allow easy integration of components in other enviroments, such as [Astro](https://astro.build/), or [Vitepress](https://vitepress.dev/), or runtime demos such as [JSBin](https://jsbin.com/).
+
+This also pairs well with [RFC#931: JS Representation of Template Tag](https://github.com/emberjs/rfcs/pull/931), in that both features together give folks enough to use Ember without compilation (compilation still recommended for optimizations, however).
+
+
+## Detailed design
+
+Users would import `renderComponent` from `@ember/renderer` (a pre-existing module)
+
+The interface:
+```ts
+export function renderComponent(
+  component: object,
+  {
+    owner,
+    env,
+    into,
+    args,
+  }: {
+    owner?: object;
+    env?: { document?: SimpleDocument | Document; isInteractive?: boolean; };
+    into: IntoTarget;
+    args?: Record<string, unknown>;
+  }
+): RenderResult | undefined {
+    /* ... implementation details ... */
+}
+```
+
+`RenderResult` is a subset of the currently a pre-existing `interface` from `@glimmer/interfaces`, but would be exposed as public API via the only the return type of `renderComponent`.
+
+It's shape is:
+```ts
+export interface RenderResult {
+    /**
+    * The element rendered in to 
+    */
+  parentElement(): SimpleElement;
+    /**
+    * Re-renders the component
+    */
+  rerender(options?: { alwaysRevalidate: false }): void;
```

Comment:
I would prefer to have `destroy` method available, to avoid extra imports. And have out-of-the-box render/remove solution.

Reply:
ya, that makes sense -- since the thing is already tied to ember, it may as well encapsulate that behavior